src/REGothEngine.hpp

```diff
@@ -42,27 +44,42 @@ namespace REGoth
 
       // Define engine options
       options.add_options()
-        ("a,game-assets", "Path to a Gothic or Gothic 2 installation", cxxopts::value<bs::Path>(assetsPath), "[PATH]")
+        ("a,game-assets", "Path to a Gothic or Gothic 2 installation", cxxopts::value<bs::Path>(originalAssetsPath), "[PATH]")
         ("video-x-res", "X resolution", cxxopts::value<unsigned int>(resolutionX), "[PX]")
         ("video-y-res", "Y resolution", cxxopts::value<unsigned int>(resolutionY), "[PX]")
-        ("video-fullscreen", "Run in fullscreen mode", cxxopts::value<bool>(fullscreen))
+        ("video-fullscreen", "Run in fullscreen mode", cxxopts::value<bool>(isFullscreen))
         ;
 
       // Allow game-assets to also be a positional
       options.parse_positional({"game-assets"});
     }
 
+    void verifyCLIEngineOptions()
+    {
+      // Resolve paths.
+      engineExecutablePath.makeAbsolute(bs::FileSystem::getWorkingDirectoryPath());
+      originalAssetsPath.makeAbsolute(bs::FileSystem::getWorkingDirectoryPath());
+
+      // TODO: Verify that assetsPath is a valid Gothic or Gothic 2 installation.
```

Comment:
There are methods doing exactly those two checks:

https://github.com/REGoth-project/REGoth-bs/blob/569d3fb1b5e4d842720da22ecf9bdeed1081840b/src/original-content/OriginalGameFiles.hpp#L75-L89

Reply:
Just below that function, there is a `isGameRoot` function which is protected but could be made public.

Reply:
Oh I don't have these changes yet, thanks for the heads up :). I will now commit the other fixes and do a quick rebase.